libs/commons-database/src/main/kotlin/gov/cdc/ocio/database/couchbase/CouchbaseCollection.kt

```diff
@@ -0,0 +1,90 @@
+package gov.cdc.ocio.database.couchbase
+
+import gov.cdc.ocio.database.persistence.Collection
+import com.couchbase.client.java.Scope
+import com.couchbase.client.java.json.JsonObject
+import com.google.gson.*
+import gov.cdc.ocio.database.utils.DateLongFormatTypeAdapter
+import gov.cdc.ocio.database.utils.InstantTypeAdapter
+import java.time.Instant
+import java.util.*
+
+
+/**
+ * Couchbase Collection implementation
+ *
+ * @property couchbaseCollection Collection
+ * @constructor
+ */
+class CouchbaseCollection(
+    collectionName: String,
+    private val couchbaseScope: Scope,
+    private val couchbaseCollection: com.couchbase.client.java.Collection
+): Collection {
+
+    private val gson = GsonBuilder()
+        .setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
+        .registerTypeAdapter(Date::class.java, DateLongFormatTypeAdapter())
+        .registerTypeAdapter(Instant::class.java, InstantTypeAdapter())
+        .create()
+
+    /**
+     * Execute the provided query and return the results as POJOs.
+     *
+     * @param query String?
+     * @param classType Class<T>?
+     * @return List<T>
+     */
+    override fun <T> queryItems(query: String?, classType: Class<T>?): List<T> {
+        val result = couchbaseScope.query(query)
+        val jsonObjects: List<JsonObject> = result.rowsAsObject()
+        val results = mutableListOf<T>()
+        jsonObjects.forEach {
+            // Couchbase items are received as an array, but within each array element is a field with the content.
+            // The content field name will match that of the collection name.
+            val collectionName = it.names.first()
+            val item = it.get(collectionName)
+            val obj = gson.fromJson(item.toString(), classType)
+            results.add(obj)
+        }
+        return results
+    }
+
+    /**
+     * Create an item from the provided data.
+     *
+     * @param id String
+     * @param item T
+     * @param classType Class<T>
+     * @param partitionKey String?
+     * @return Boolean
+     */
+    override fun <T> createItem(id: String, item: T, classType: Class<T>, partitionKey: String?): Boolean {
+        val upsertResult = couchbaseCollection.upsert(
+            id,
+            JsonObject.fromJson(gson.toJson(item, classType))
+        )
```

Comment:
Answered this earlier, but the partitionKey is a necessary part of the interface, but not used by all implementations of that interface - only cosmos.  I added it as an enhancement idea to our ticket so we can re-examine.  It might make sense for this to be passed in as a param (since it never changes) to cosmos and remove it from the interface.